printer/src/algorithm.rs

```diff
@@ -99,9 +101,14 @@ impl Printer {
             print_stack: Vec::new(),
             indent: 0,
             pending_indentation: 0,
+            dos_line_endings: false,
         }
     }
 
+    pub fn set_dos(&mut self, value: bool) {
```

Comment:
we could pass this to the instructor of `Printer` instead, no need for a setter in that case.
(I don't think you would ever switch to a different line ending style halfway formatting)